bin/weewx/drivers/vantage.py

```diff
@@ -478,120 +518,182 @@ def genDavisLoopPackets(self, N=1):
+                    if loop > 0:
+                        # ... decode it
+                        loop_packet = self._unpackLoopPacket(_buffer[:95])
+                        # Yield it
+                        yield loop_packet
+
+                    if loop == N:
+                        syslog.syslog(syslog.LOG_DEBUG, "vantage: genDavisLoopPackets; readloop complete!!!")
+                        _readloop = False
+                    else:
+                        ntries = 1
+                else:
+                    syslog.syslog(syslog.LOG_DEBUG, "vantage: genDavisLoopPackets; length loop_packet %s; expcted 99" %
+                                  len(_buffer))
+                    syslog.syslog(syslog.LOG_DEBUG, "vantage: genDavisLoopPackets; restart LOOP 200")
+                    self.port.closePort()
+                    self.port.openPort()
+                    break
+
     def genArchiveRecords(self, since_ts):
         """A generator function to return archive packets from a Davis Vantage station.
-        
+
         since_ts: A timestamp. All data since (but not including) this time will be returned.
         Pass in None for all data
-        
+
         yields: a sequence of dictionaries containing the data
         """
-        
-        if since_ts:
-            since_tt = time.localtime(since_ts)
-            # NB: note that some of the Davis documentation gives the year offset as 1900.
-            # From experimentation, 2000 seems to be right, at least for the newer models:
-            _vantageDateStamp = since_tt[2] + (since_tt[1]<<5) + ((since_tt[0]-2000)<<9)
-            _vantageTimeStamp = since_tt[3] *100 + since_tt[4]
-            syslog.syslog(syslog.LOG_DEBUG, 'vantage: Getting archive packets since %s' % weeutil.weeutil.timestamp_to_string(since_ts))
-        else:
-            _vantageDateStamp = _vantageTimeStamp = 0
-            syslog.syslog(syslog.LOG_DEBUG, 'vantage: Getting all archive packets')
-     
-        #Pack the date and time into a string, little-endian order
-        _datestr = struct.pack("<HH", _vantageDateStamp, _vantageTimeStamp)
-        
-        # Save the last good time:
-        _last_good_ts = since_ts if since_ts else 0
-        
-        # Try to retrieve the starting page and index:
-        for n in range(self.max_tries):
-            try:
-                # Wake up the console...
-                self.port.wakeup_console(self.max_tries, self.wait_before_retry)
-                # ... request a dump...
-                self.port.send_data('DMPAFT\n')
-                # ... from the designated date (allow only one try because that's all the console allows):
-                self.port.send_data_with_crc16(_datestr, max_tries=1)
-                
-                # Get the response with how many pages and starting index and decode it. Again, allow only one try:
-                _buffer = self.port.get_data_with_crc16(6, max_tries=1)
+        _catchup = True
```

Comment:
Don't know, Tom. It solved the problem for the serial device. Not sure if close/openPort is needed for the ip version too.  I will check and report.